pkg/quota/v1/evaluator/core/services.go

```diff
@@ -120,22 +122,39 @@ func (p *serviceEvaluator) Usage(item runtime.Object) (corev1.ResourceList, erro
 	// default service usage
 	result[serviceObjectCountName] = *(resource.NewQuantity(1, resource.DecimalSI))
 	result[corev1.ResourceServices] = *(resource.NewQuantity(1, resource.DecimalSI))
-	result[corev1.ResourceServicesLoadBalancers] = resource.Quantity{Format: resource.DecimalSI}
-	result[corev1.ResourceServicesNodePorts] = resource.Quantity{Format: resource.DecimalSI}
+	// check nodeport and load balancer only if the runtime svc is one of them.
 	switch svc.Spec.Type {
 	case corev1.ServiceTypeNodePort:
 		// node port services need to count node ports
 		value := resource.NewQuantity(int64(ports), resource.DecimalSI)
 		result[corev1.ResourceServicesNodePorts] = *value
 	case corev1.ServiceTypeLoadBalancer:
```

Comment:
do we also need to count usage by HealthCheckNodePort?

Reply:
I'm not counting this yet. 
Should add 1 when `type is set to LoadBalancer and externalTrafficPolicy is set to Local`

Reply:
IMHO `HealthCheckNodePort` != `NodePort` , and as a user it may be confusing to have to consider them equal for quotas, but let's ask @thockin for his opinion

Reply:
It seems that HealthCheckNodePort is not taken into account in previous implementations. I'm not sure if we should count like below. 
```
diff --git a/pkg/quota/v1/evaluator/core/services.go b/pkg/quota/v1/evaluator/core/services.go
index ae5504a6eca..45b07526b59 100644
--- a/pkg/quota/v1/evaluator/core/services.go
+++ b/pkg/quota/v1/evaluator/core/services.go
func portsWithNodePorts(svc *corev1.Service) *resource.Quantity {
	count := 0
+	if svc.Spec.externalTrafficPolicy == api.ServiceExternalTrafficPolicyTypeLocal {
+		count++
+	}
	for _, p := range svc.Spec.Ports {
		if p.NodePort != 0 {
			count++
		}
	}
	return resource.NewQuantity(int64(count), resource.DecimalSI)
}

```

Reply:
We probably want to backport this fix (at least the zero-value portion of it). Let's get this in as-is, then follow-up with including the health check node port in quota in a separate PR

Reply:
OK, I will test HealthCheckNodePort behavior tomorrow. Nice catch.